lib/services/causes_service.dart

```diff
@@ -60,14 +62,15 @@ class CausesService {
   Stream<CausesUser?> get userInfoStream => _userInfoStore.userInfoStream;
 
   List<Cause>? _causes;
-  Future<List<Cause>> _fetchCauses() async {
+
+  Future<List<Cause>?> _fetchCauses() async {
     _logger.info('Fetching causes from the service');
     final (response, userInfo) = await (
       _causeServiceClient.getCausesApi().causesList(),
       getUserInfo(),
     ).wait;
-    return response.data!
-        .map(
+    return response.data!.results
+        ?.map(
```

Comment:
I would say we can just do ! here as well. I don't know why the types are optional here

Reply:
No this is wrong, your thing is better we should probably add ? for data